pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -175,7 +176,24 @@ func (r *Reader) Read(dst []byte) (int, error) {
 	}
 
 	// for appendCount==0, the following code would add `0` to fingerprint
-	r.Fingerprint.FirstBytes = append(r.Fingerprint.FirstBytes[:r.Offset], dst[:appendCount]...)
+	if r.Fingerprint.FirstBytes == nil {
+		r.Fingerprint.FirstBytes = dst[:appendCount]
+	} else {
+		r.Fingerprint.FirstBytes = append(r.Fingerprint.FirstBytes[:r.Offset], dst[:appendCount]...)
+	}
+	if r.Fingerprint.HashInstance == nil {
+		h := fnv.New64()
+		h.Write(r.Fingerprint.FirstBytes)
+		hashed := h.Sum64()
+		r.Fingerprint.HashInstance = &h
+		r.Fingerprint.HashBytes = hashed
+	} else {
+		hashInstance := *r.Fingerprint.HashInstance
+		hashInstance.Write(r.Fingerprint.FirstBytes[r.Fingerprint.BytesUsed:len(r.Fingerprint.FirstBytes)])
+		r.Fingerprint.HashBytes = hashInstance.Sum64()
+	}
+	r.Fingerprint.BytesUsed = len(r.Fingerprint.FirstBytes)
```

Comment:
I don't think we should have to worry about whether the struct has been properly instantiated at this point. If `HashBytes` and `BytesUsed` are in fact extraneous as I suspect, then I think we could reduce this to:
```suggestion
	r.Fingerprint.HashInstance.Write(r.Fingerprint.FirstBytes[dst[:appendCount])
```